dataservices/dataservices.go

```diff
@@ -0,0 +1,239 @@
+package main
```

Comment:
Might want to pull out some of this out of the main package into a dataservices package.  The file with the main package should be tiny, I think. Basically, the main function that calls into the dataservices package to do everything else.